src/apps/components/run-header.js

```diff
@@ -54,15 +54,28 @@ class RunHeader extends React.Component {
 
         let query = !this.props.bundle ? { local: true } : this.props.location.query;
 
+        var codeBtn;
+        if (this.props.runMode.path || this.props.runMode.rendezvous || this.props.runMode.local) {
+            codeBtn = (<button
+                onClick={this._toggleShowJuttle}
+                className={showJuttleClasses}>
+                <i className="fa fa-lg fa-fw fa-code"></i>
+                <div className="font-btn-name">code</div>
+            </button>);
+        } else {
+            codeBtn = (<Link
+                onClick={this._toggleShowJuttle}
+                to={{ pathname: '/', query: query}}
```

Comment:
yeah sure Vlad, i like doing it for the contents. We'll still need the ugly branch logic for the outer component though